pkg/controller/controller_utils.go

```diff
@@ -542,6 +544,39 @@ func GetPodFromTemplate(template *v1.PodTemplateSpec, parentObject runtime.Objec
 	return pod, nil
 }
 
+func (r RealPodControl) validateNodeConflict(pod *v1.Pod) error {
+	if len(pod.Spec.NodeName) == 0 {
+		return nil
+	}
+
+	node, err := r.KubeClient.Core().Nodes().Get(pod.Spec.NodeName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	nodeInfo := schedulercache.NewNodeInfo()
+	err = nodeInfo.SetNode(node)
+	if err != nil {
+		return err
+	}
+
+	fit, reasons, err := predicates.PodSelectorMatches(pod, nil, nodeInfo)
```

Comment:
@kargakis/@xiangpengzhao , `Pending` maybe some problem as kubelet did not know "who" set the hostname: if scheduler set it, there's no chance for scheduler to "correct" it.

I also doubt whether controller has enough knowledge for that; does it make sense to add an admit controller (for default scheduler) to check this?